apps/heft/src/plugins/TypeScriptPlugin/Tslint.ts

```diff
@@ -0,0 +1,152 @@
+import { IExtendedSourceFile, IExtendedProgram } from './internalTypings/TypeScriptInternals';
+import { Terminal, JsonFile, Colors } from '@rushstack/node-core-library';
+import { IExtendedFileSystem } from '../../utilities/fileSystem/IExtendedFileSystem';
+import { ResolveUtilities } from '../../utilities/ResolveUtilities';
+import { IExtendedLinter } from './internalTypings/TslintInternals';
+
+interface ITslintOptions extends ILinterBaseOptions {
+  tslintPackagePath: string;
+
+  fileSystem: IExtendedFileSystem;
+}
+
+export class Tslint extends LinterBase<TTslint.RuleFailure> {
+  private readonly _tslint: typeof TTslint;
+  private readonly _fileSystem: IExtendedFileSystem;
+
+  private _tslintConfiguration: TTslint.Configuration.IConfigurationFile;
+  private _linter: IExtendedLinter;
+  private _enabledRules: TTslint.IRule[];
+  private _ruleSeverityMap: Map<string, TTslint.RuleSeverity>;
+  protected _lintResult: TTslint.LintResult;
+
+  public constructor(options: ITslintOptions) {
+    super('tslint', options);
+
+    this._tslint = require(options.tslintPackagePath);
+    this._fileSystem = options.fileSystem;
+  }
+
+  public static getConfigHash(
+    configFilePath: string,
+    terminal: Terminal,
+    fileSystem: IExtendedFileSystem
+  ): crypto.Hash {
+    interface IMinimalConfig {
+      extends?: string;
+    }
+
+    terminal.writeVerboseLine(`Examining config file "${configFilePath}"`);
+
+    const rawConfig: string = fileSystem.readFile(configFilePath);
+    const parsedConfig: IMinimalConfig = JsonFile.parseString(rawConfig);
+    let hash: crypto.Hash;
+    if (parsedConfig.extends) {
+      const extendsFullPath: string = ResolveUtilities.resolvePackagePath(
+        parsedConfig.extends,
+        path.dirname(configFilePath)
+      );
+      hash = Tslint.getConfigHash(extendsFullPath, terminal, fileSystem);
+    } else {
+      hash = crypto.createHash('sha1').update(rawConfig);
+    }
+
+    return hash.update(rawConfig);
+  }
+
+  public printVersionHeader(): void {
+    this._terminal.writeLine(`Using TSLint version ${this._tslint.Linter.VERSION}`);
+  }
+
+  public reportFailures(): void {
+    if (this._lintResult.failures?.length) {
+      this._terminal.writeWarningLine(
+        `Encountered ${this._lintResult.failures.length} TSLint error${
+          this._lintResult.failures.length > 1 ? 's' : ''
+        }:`
+      );
+      for (const tslintFailure of this._lintResult.failures) {
+        const buildFolderRelativeFilename: string = path.relative(
+          this._buildFolderPath,
+          tslintFailure.getFileName()
+        );
+        const { line, character } = tslintFailure.getStartPosition().getLineAndCharacter();
+        const severity: string = tslintFailure.getRuleSeverity().toUpperCase();
+        this._terminal.writeWarningLine(
+          '  ',
+          Colors.yellow(`${severity}: ${buildFolderRelativeFilename}:${line + 1}:${character + 1}`),
```

Comment:
BTW I noticed that the API Extractor output is still using `filename.ts(123,234)` style formatting. This is easy to fix because `ExtractorMessage` exposes the individual parts:    `sourceFilePath`, `sourceFileLine`, `sourceFileColumn`, `formatMessageWithoutLocation()`

Reply:
Or, if Heft is blazing the trail for a new standard, we could change API Extractor's default formatting.

Reply:
Here's a fix for that: https://github.com/microsoft/rushstack/pull/2009